### ISSUE TITLE
Memory leakage 

### ISSUE THREAD

**Issue by MohannadEhabBarakat** [MohannadEhabBarakat/U-2-Net#1]
**Description:**
During training, the validation fails. On the GPU it crashes with an insufficient memory error. On CPU it runs for 19 iterations then stops with a message "killed". 

**OS:** Ubuntu
**GPU:** 1080 ti
**memory:** 64 Gb

**Comment by MohannadEhabBarakat** [comment]
This happened because of an undesired function of pytorch. When using a variable with backpropagation enabled in calculations the result is also backpropagation enabled. This accumulates a new node into the computing graph each iteration. This results in a growing graph instead of just running an equation. 

There is tow ways to fix this:

1) **If you need the result temporarily** 
Like in this example
`d0, d1, d2, d3, d4, d5, d6 = net(inputs_v)
 loss2, loss = muti_bce_loss_fusion(d0, d1, d2, d3, d4, d5, d6, labels_v)
`
d0, d1, d2, d3, d4, d5, d6, loss2, loss are not needed after plotting with TensorBoard or wandb.

In such case just use `del`
`del d0, d1, d2, d3, d4, d5, d6, loss2, loss`

2) **if you need the result**
So during the calculation of IoU, the IoU of each iteration should be accumulated to calculate the mean of each epoch. In such a case use `variable.data.item()` this we return the value without growing graph.
